torch/_higher_order_ops/scan.py

```diff
@@ -48,6 +48,20 @@ def _extract_carry_and_out(flat_out: List[Any], num_carry: int):
     return flat_out[:num_carry], flat_out[num_carry:]
 
 
+# An custom op that's used for lowering scan in inductor.
+# Compared with aten.select, the lowering rule for this is more specialized
+# to the scan operator. For example, we skips a bunch of checks that we're
+# certain is true for scan.
+@torch.library.custom_op("_scan_helper::unsafe_select", mutates_args=())  # type: ignore[misc]
```

Comment:
Nice to use unsafe_select, but could we just register aten._unsafe_select in the same way that we have the other aten unsafe operators ? there is nothing specific about scan to this.

Reply:
yeah, sounds good.